pylint/checkers/typecheck.py

```diff
@@ -523,6 +525,77 @@ def _emit_no_member(
     return True
 
 
+def _get_all_attribute_assignments(
+    node: astroid.FunctionDef, name: str | None = None
+) -> set[str]:
+    attributes = set()
+    for child in node.nodes_of_class((astroid.Assign, astroid.AnnAssign)):
+        targets = []
+        if isinstance(child, astroid.Assign):
+            targets = child.targets
+        elif isinstance(child, astroid.AnnAssign):
+            targets = [child.target]
+        for assign_target in targets:
+            if not isinstance(assign_target, astroid.AssignAttr):
+                continue
+            if not isinstance(assign_target.expr, astroid.Name):
+                continue
+            if name is None or assign_target.expr.name == name:
+                attributes.add(assign_target.attrname)
+    return attributes
+
+
+def _enum_has_attribute(
+    owner: astroid.Instance | nodes.ClassDef, node: nodes.Attribute
```

Comment:
We should probably import this from `objects` instead of `astroid`.

Reply:
Sorry, don't understand what you're talking about.

Assuming you're referencing the type hints (albeit unsure which exact one), I copied the existing style in the file:
* `Instance`: mostly imported from `astroid`
* `ClassDef`/`Attribute`: mostly imported from `nodes`

Reply:
Ah okay, then the style on the rest of the file is messed up sadly. `Instance` comes from `astroid.objects` but is also imported in the ``__init__`` of `astroid`.

Let's keep as is then!